pkg/validation/internal/good_practices.go

```diff
@@ -0,0 +1,76 @@
+package internal
+
+import (
+	goerrors "errors"
+	"fmt"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+// GoodPracticesValidator validates the bundle against criteria and suggestions define as
+// good practices for bundles under the operator-framework solutions
+//
+// This validator will raise an WARNING when:
+//
+// - The resources request for CPU and/or Memory are not defined for any of the containers found in the CSV
+var GoodPracticesValidator interfaces.Validator = interfaces.ValidatorFunc(goodPracticesValidator)
+
+func goodPracticesValidator(objs ...interface{}) (results []errors.ManifestResult) {
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateGoodPracticesFrom(v))
+		}
```

Comment:
Should we warn or panic on the default case?

Reply:
That is true. We have not in any of the validators.
Tried to cover all scenarios but that seems out of the scope.

Reply:
No, because that would mean panic or warning when we are checking all objects looking for the bundle. (see that it is a for)

To ensure that we always will have the bundle here it is checked before in the default checks made.

Reply:
ah ok - you're right! I get it now.